### ISSUE TITLE
vite4 `defineConfig()` does not accept `test` props as valid

### ISSUE THREAD

**Issue by unional** [vitest-dev/vitest#2486]
### Describe the bug

```sh
Argument of type '{ plugins: (PluginOption[] | { enforce: "pre"; api?: any; cacheKey?: string | undefined; name: string; augmentChunkHash?: ObjectHook<(this: PluginContext, chunk: RenderedChunk) => string | void, {}> | undefined; ... 24 more ...; watchChange?: ObjectHook<...> | undefined; })[]; build: { ...; }; resolve: { ...; }; tes...' is not assignable to parameter of type 'UserConfigExport'.
  Object literal may only specify known properties, and 'test' does not exist in type 'UserConfigExport'.ts(2345)
```

When using vite 4.

Seems like need to update the declaration merging.

### Reproduction

gh repo clone justland/just-web-react
git checkout vite4
pnpm i

check out any of the `vite.config.ts` file.

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 36.75 GB / 63.92 GB
  Binaries:
    Node: 18.7.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.15.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.819.0), Chromium (108.0.1462.42)
    Internet Explorer: 11.0.22621.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Check you lock file. You have several Vitest versions installed.

**Comment by sheremet-va** [comment]
It seems like pnpm prefers lower version, because it's already installed and allowed by Vitest. Maybe we should just drop Vite 3 support, and use `^4.0.0` in dependencies.

**Comment by unional** [comment]
Yes you are correct. It could also due to ordering. When I run `pnpm up -r vite` again (after `vitest` and `@vitejs/coverage-c8), it does pick up 4.0.0 and fixed the issue. Thanks!
